app/models/scc_account.rb

```diff
@@ -187,18 +189,30 @@ def update_scc_repositories(upstream_repositories)
       cached_repository.installer_updates = ur['installer_updates']
       # should be called after all attributes are set in case of dependencies (currently: description)
       cached_repository.name = cached_repository.pretty_name
+      cached_repository.subscription_valid = true
       cached_repository.save!
       upstream_repo_ids << ur['id']
+      # set invalidated record to true, if exists
+      invalidated_repos = revalidate_subscription_status(invalidated_repos, ur[id])
     end
     ::Foreman::Logging.logger('foreman_scc_manager').debug "Found #{upstream_repo_ids.length} repositories"
-    # delete repositories beeing removed upstream
+
+    # all scc repos that are kept but not available upstream need to be marked invalid
+    # subscription_valid can be set to nil
+    to_invalidate = invalidated_repos.select { |ir| !ir.katello_root_repository_id.nil? && !ir.subscription_valid }
+    invalidate_subscription_status(to_invalidate, true)
+    ::Foreman::Logging.logger('foreman_scc_manager').debug "Invalidating #{to_invalidate.count} expired repositories"
```

Comment:
log message should occur before method call.

Reply:
Fixed.